File: code-coaches/medium/snowballing-numbers/snowballing_numbers.rb
def is_each_number_greater_than_the_sum(nums)
  nums.each_with_index do |num, idx|
    return false if num < nums[0...idx].sum
  end
  true
end

def main
  nums_length = gets.chomp.to_i
  nums = []
  nums_length.times do
    nums << gets.chomp.to_i
  end

  puts is_each_number_greater_than_the_sum(nums)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/snowballing-numbers/snowballing_numbers.py
def is_each_number_greater_than_the_sum(nums: list[int]) -> bool:
    for idx, num in enumerate(nums):
        if num < sum(nums[:idx]):
            return False

    return True


def main():
    nums_length = int(input())
    nums = [int(input()) for _ in range(nums_length)]
    if is_each_number_greater_than_the_sum(nums):
        print("true")
    else:
        print("false")


if __name__ == "__main__":
    main()
